bid-distribution/src/settlement_claims.rs

```diff
@@ -0,0 +1,112 @@
+use crate::sam_meta::ValidatorSamMeta;
+use log::info;
+use settlement_engine::stake_meta_index::StakeMetaIndex;
+use crate::settlement_config::SettlementConfig;
+use solana_sdk::native_token::sol_to_lamports;
+use solana_sdk::pubkey::Pubkey;
+use std::collections::HashMap;
+use std::cmp::min;
+use settlement_engine::settlement_claims::{SettlementReason, Settlement, SettlementClaim, SettlementCollection};
+
+pub fn generate_bid_settlement_collection(
+    stake_meta_index: &StakeMetaIndex,
+    sam_validator_metas: &Vec<ValidatorSamMeta>,
+    stake_authority_filter: &dyn Fn(&Pubkey) -> bool,
+    settlement_config: &SettlementConfig,
+) -> SettlementCollection {
+    let settlements = generate_bid_settlements(&stake_meta_index, &sam_validator_metas, &stake_authority_filter, &settlement_config);
```

Comment:
I was thinking about it, but it will not be as straight forward as there. I guess I can extract epoch from sam_validator_metas vec and compare it with stake_meta epoch.